### PR TITLE
feat: Verify data in tests (row count and first row)

### LINKED ISSUE #16: Verify Data in Tests
We current have a bunch of failures from the integration tests testing the data. After we fix the failures that are happening, then we need to be sure the data that is in the DB is correct. https://github.com/MeltanoLabs/target-sqlite is where I took most of the tests from, and luckily there are already data test there written using SQL alchemy. We should take these. 

- [x] xfail everything that's failing with issues for each

### PR DESCRIPTION
**Pull request by sebastianswms**
Implement a method to verify data in tests. Currently only checks the total number of rows in a table and the data in the first row. I believe this provides good test coverage while keeping things manageable.

Closes #16

### CODE REVIEW

**Comment by sebastianswms** [comment]
#219 and #220 both add new pytests, so additions will need to be made to this PR to verify their data.

File: target_postgres/tests/test_standard_target.py [hunk 1]
```diff
@@ -116,6 +117,42 @@ def remove_metadata_columns(row: dict) -> dict:
     return new_row
 
 
+def verify_data(
```
Comment (visch) [inline target_postgres/tests/test_standard_target.py:120]: Love the simplicity of this. As a part of this PR (I know this expands the scope a bit I'd be happy if you want to split this to a seperate PR) can we also be sure we're pulling in the SDK tests that have been added. Honestly another file probably makes the most sense (maybe sdk_tests.py or something) See https://github.com/MeltanoLabs/target-snowflake/blob/main/tests/test_target_snowflake.py#L71

File: target_postgres/tests/test_standard_target.py [hunk 2]
```diff
@@ -242,101 +280,49 @@ def test_schema_no_properties(postgres_target):
...
+    user = {"id": 1, "name": "Johny"}
+    location = {"id": 1, "name": "Philly"}
+    user_in_location = {
+        "id": 1,
+        "user_id": 1,
+        "location_id": 4,
+        "info": {"weather": "rainy", "mood": "sad"},
+    }
 
-    with engine.connect() as connection:
```
Comment (visch) [inline target_postgres/tests/test_standard_target.py:267]: I like that you want to consolidate the Tests but I think this test had a pretty solid data test and you removed a lot of the checks that were there before. We could just leave this one alone, or you could expand your verify data function to be able to work with multiple rows of data.

**Review by visch — CHANGES_REQUESTED** [review]

File: target_postgres/tests/test_standard_target.py (hunk 1 above)
Comment (sebastianswms) [reply target_postgres/tests/test_standard_target.py:120]: Created #223

File: target_postgres/tests/test_standard_target.py (hunk 2 above)
Comment (sebastianswms) [reply target_postgres/tests/test_standard_target.py:277]: Expanded to work with multiple rows.

File: target_postgres/tests/test_standard_target.py [hunk 3]
```diff
@@ -116,6 +117,57 @@ def remove_metadata_columns(row: dict) -> dict:
...
+        if primary_key is not None and check_data is not None:
+            if isinstance(check_data, dict):
+                result = connection.execute(
+                    sqlalchemy.text(
+                        f"SELECT * FROM {full_table_name} ORDER BY {primary_key}"
+                    )
+                )
+                assert result.rowcount == number_of_rows
+                result_dict = remove_metadata_columns(result.first()._asdict())
+                assert result_dict == check_data
```
Comment (visch) [inline target_postgres/tests/test_standard_target.py:150]: I'm impressed with Python's dictionary comparisons!

**Review by visch — APPROVED** [review]

**Pull request merged**